packages/valory/skills/apy_estimation_abci/tests/test_models.py

```diff
@@ -127,6 +130,45 @@ def test_setup(
 class TestAPYParams:
     """Test `APYParams`"""
 
+    @staticmethod
+    @given(end=st.one_of(st.none(), st.integers()), ts_length=st.integers())
+    @settings(deadline=None, database=database.InMemoryExampleDatabase())
+    def test_start(end: Optional[int], ts_length: int) -> None:
+        """Test `start` property."""
+        args = APY_PARAMS_ARGS
+        # TypedDict can’t be used for specifying the type of a **kwargs argument: https://peps.python.org/pep-0589/
+        kwargs: dict = deepcopy(APY_PARAMS_KWARGS)  # type: ignore
+        kwargs["history_end"] = end
+        params = APYParams(*args, **kwargs)
+
+        expected = None if end is None else end - ts_length
+
+        with mock.patch.object(
+            APYParams,
+            "ts_length",
+            new_callable=mock.PropertyMock,
+            return_value=ts_length,
+        ):
+            if expected is None:
+                assert params.start is expected
+            else:
+                assert params.start == expected
```

Comment:
`assert params.start == expected`  will work for both cases

Reply:
Done in e8fd98314e7d854e9e763fe1ecb944ea7ebac9a6.